src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoNullable.cs

```diff
@@ -79,6 +81,46 @@ protected override void OnWrite(ref WriteStackFrame current, Utf8JsonWriter writ
             }
         }
 
+        protected override void OnWriteDictionary(ref WriteStackFrame current, Utf8JsonWriter writer)
```

Comment:
Do we need to make similar changes to other `JsonPropertyInfoCommon<TClass, TDeclaredProperty, TRuntimeProperty, TConverter>` types? like `JsonPropertyInfoNotNullable` and `JsonPropertyInfoNotNullableContravariant` or are they not relevant here since they are explicitly about non-nullable types?

Reply:
No we don't. 

> are they not relevant here since they are explicitly about non-nullable types?

Precisely. The logic in `JsonSerializer.WriteDictionary()` is suited for non-nullable values. The reason for the bug this change fixes is that we didn't have an implementation for (and specific to) nullable values.